### PR TITLE
build: Support Go 1.13.x

### PR DESCRIPTION
**Pull request by dio**
## Description

- This supports building the project using Go 1.13. 
- This adds `REGISTRY` param in `Makefile` to allow overriding pushing images to a different registry.

## Motivation and Context

- To stay up-to-date to the latest tooling.
- Allow pushing images to a different repo.

## How Has This Been Tested?

- It passes the CI (against my fork).
- I did `$ make docker-build-all` locally.
- I did `$ REGISTRY=dio123 make docker-push-all` locally to push to my own repo.

## Checklist:

- [x] My change requires a change to the documentation or CHANGELOG.
- [x] I have updated the documentation/CHANGELOG accordingly.
- [x] I have created a feature (non-master) branch for my PR.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,6 +1,8 @@
 include .env
 BINARY := oauth2_proxy
 VERSION := $(shell git describe --always --dirty --tags 2>/dev/null || echo "undefined")
+# Allow to override image repository.
+HUB ?= quay.io/pusher
```
Comment (steakunderscore) [inline Makefile]: nit: I would have used `REGISTRY` rather than `HUB`.

Comment (dio) [reply Makefile]: OK, done in 29f3b96. Thank you for the review!

File: Makefile [hunk 2]
```diff
@@ -27,31 +29,31 @@ $(BINARY):
 
 .PHONY: docker
 docker:
-	docker build -f Dockerfile -t quay.io/pusher/oauth2_proxy:latest .
+	docker build -f Dockerfile -t $(HUB)/oauth2_proxy:latest .
```
Comment (JoelSpeed) [inline Makefile]: You'll need to find and replace `HUB` as part of the renaming

Comment (dio) [reply Makefile]: 🤦🏽‍♂️I thought I have done that. Sorry. Fixed.

**Pull request merged**